xoa/data_source_hosts.go

```diff
@@ -0,0 +1,86 @@
+package xoa
+
+import (
+	"github.com/ddelnano/terraform-provider-xenorchestra/client"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"log"
+)
+
+func dataSourceXoaHosts() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceHostsRead,
+		Schema: map[string]*schema.Schema{
+			"master": &schema.Schema{
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"hosts": &schema.Schema{
+				Type:     schema.TypeList,
+				Computed: true,
+				Elem: &schema.Schema{
+					Type: schema.TypeMap,
+					Elem: &schema.Schema{
+						Type: schema.TypeString,
+					},
```

Comment:
Rather than this just being a string we could use [dataSourceXoaHost](https://github.com/terra-farm/terraform-provider-xenorchestra/blob/6d0b85c152094efd9ee16ad0c8b6a080b701c305/xoa/data_source_host.go#L10) which would give us access to all the attributes that the individual host data source supports. Right now that is just `name_label` and `id` but that could include CPUs, hostnames, memory, XCP version, etc in the future

Reply:
> Rather than this just being a string we could use [dataSourceXoaHost](https://github.com/terra-farm/terraform-provider-xenorchestra/blob/6d0b85c152094efd9ee16ad0c8b6a080b701c305/xoa/data_source_host.go#L10) which would give us access to all the attributes that the individual host data source supports. Right now that is just `name_label` and `id` but that could include CPUs, hostnames, memory, XCP version, etc in the future

I tried really hard to return a slice of `host`structs  from GetHostsByPoolName() and reuse the dataSourceXoaHost to set the `hosts` based on the `host`schema. Terraform won't transform the results as needed.  This always results in an error telling me :

`Invalid address to set: []string{"hosts", "0", "Id"}`

I am a little stuck here. Need to do a little more research and debugging. Maybe you have a quick solution for this error?

Reply:
I wasn't able to reproduce the Invalid address to set: []string{"hosts", "0", "Id"} issue you mentioned above. However, I was able to get the tests passing and ensure that the hosts attribute was properly set. You can see the changes I made on pr-review-add-hosts-data-source branch (in b69c226b1a2af98c22ae8ca4020810dbcea90127).

Let me know if that doesn't address the issue you were seeing. I ran the test with the following command: 
`TF_LOG=debug TEST=TestAccXenorchestraDataSource_hosts make testacc`